user-interface/src/main/java/life/qbic/datamanager/views/projects/create/AddProjectDialog.java

```diff
@@ -153,23 +157,22 @@ private void onCancelClicked() {
   }
 
   private void onConfirmClicked(ClickEvent<Button> event) {
-    if (projectDesignLayout.validate().isInvalid()) {
+    if(projectDesignLayout.validate().isInvalid()){
       return;
     }
-    if (fundingInformationLayout.validate().isInvalid()) {
+    projectCreationInformation.setProjectDesign(projectDesignLayout.getProjectDesign());
+    if(fundingInformationLayout.validate().isInvalid()){
       return;
     }
-    if (collaboratorsLayout.validate().isInvalid()) {
+    projectCreationInformation.setFundingEntry(fundingInformationLayout.getFundingInformation());
+    if(collaboratorsLayout.isInvalid()){
       return;
     }
+    projectCreationInformation.setProjectCollaborators(collaboratorsLayout.getProjectCollaborators());
     if (experimentalInformationLayout.validate().isInvalid()) {
-      return;
+        return;
     }
-    fireEvent(new ConfirmEvent(this, projectDesignLayout.getProjectDesign(),
-        fundingInformationLayout.getFundingInformation(),
-        collaboratorsLayout.getProjectCollaborators(),
-        experimentalInformationLayout.getExperimentalInformation(), true));
-
+    fireEvent(new ConfirmEvent(this, projectCreationInformation, experimentalInformationLayout.getExperimentalInformation(), true));
```

Comment:
You could construct an immutable record here. I do not see the advantage of the mutable object you fill.

Reply:
Valid Point, i thought it's easier to understand since the object is filled step by step but it doesn't matter in the end